### PR TITLE
Added Functionality for checkmate & more

### PR DESCRIPTION
**Pull request by tuura01**
Added functionality for checkmate #12 
Added functionality for checkmate & check in makeMove.c.
Added functionality for pinning pieces in makeMove.c

 Game will now end if one player mates the other.
 Game will warn player about check.
 Game will not allow you to make a move that doesnt resolve the current check
 Game will not allow you to make moves that will cause you to self check.

### CODE REVIEW

File: tests/unitTests/utCheckmate.c [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#include <stdio.h>
+#include <checkmate.h>
+#include <makeMove.h>
+#include <assert.h>
+
+static void baseBoard(char * board[8][8]);
+
+history * hs = NULL;
+
+static void haha(move *mx);
```
Comment (Linkan42) [inline tests/unitTests/utCheckmate.c:10]: Maybe not the best name for a method 😉

**Review by Linkan42 — APPROVED** [review]
Nice! Tested it myself quickly and it seems to work as expected 👍

**Comment by mosmar99** [comment]
Looks good to me as well, besides that method name. 
Good job.

**Comment by tuura01** [comment]
Thank you for your reviews, i have updated the method name. Merging this branch and closing #12 .

**Pull request merged**